frontend/packages/dev-console/src/components/topology/topology-utils.ts

```diff
@@ -228,7 +228,12 @@ const getTopologyEdgeItems = (
             sbr.spec.backingServiceSelector.resourceRef &&
           _.get(deployment, 'metadata.ownerReferences[0].kind') ===
             sbr.spec.backingServiceSelector.kind;
-        return targetFound;
+        const appGroup = _.get(
+          targetFound,
```

Comment:
@karthikjeeyar why are you using `targetFound` to get the `appGroup` and how is it even working? `targetFound` is a boolean and when it is `true` it indicates that the corresponding deployment is the edge target and then we get the `uid` from its metadata. I will suggest you to use the corresponding `deployment` to get the app-group when `targetFound` is true.

Reply:
Its probably working because you are returning `null` when the desired property is not found. In this case `appGroup` will always be `null`.

Reply:
@divyanshiGupta Oops, It has to be deployment. Changed it.